app/src/main/java/org/mozilla/fenix/components/Core.kt

```diff
@@ -218,6 +219,18 @@ class Core(
                     .periodicallyInForeground(interval = 30, unit = TimeUnit.SECONDS)
                     .whenGoingToBackground()
                     .whenSessionsChange()
+
```

Comment:
^^ This whole block we should refactor to a new `RestoreUseCase` (or similar) in A-C: https://github.com/mozilla-mobile/android-components/issues/8600

Reply:
This would also make it testable which is very hard to do here right now.